eras/babbage/impl/src/Cardano/Ledger/Babbage/Translation.hs

```diff
@@ -180,6 +181,19 @@ translateTxOut (Alonzo.TxOut addr value dh) = TxOut addr value d SNothing
       SNothing -> NoDatum
       SJust d' -> DatumHash d'
 
+-- | A word is 8 bytes, so to convert from coinsPerUTxOWord to coinsPerUTxOByte,
+-- we divide by 8, after first adding 7 so that we are rounding up.
+coinsPerUTxOWordToCoinsPerUTxOByte :: Coin -> Coin
+coinsPerUTxOWordToCoinsPerUTxOByte (Coin c) = Coin $ c + 7 `div` 8
+
 translatePParams ::
-  Alonzo.PParams' f (AlonzoEra c) -> PParams' f (BabbageEra c)
-translatePParams (Alonzo.PParams {..}) = PParams {..}
+  Alonzo.PParams (AlonzoEra c) -> PParams (BabbageEra c)
+translatePParams (Alonzo.PParams {_coinsPerUTxOWord = cpuw, ..}) = PParams {_coinsPerUTxOByte = cpub, ..}
+  where
+    cpub = coinsPerUTxOWordToCoinsPerUTxOByte cpuw
+
+translatePParamsUpdate ::
+  Alonzo.PParamsUpdate (AlonzoEra c) -> PParamsUpdate (BabbageEra c)
```

Comment:
I couldn't find a nice way to keep `translatePParams` parametric in `HKD f`, (where `f` is either `Identity` or `StrictMaybe`). Maybe someone has a suggestion?

Reply:
Not sure I am a fan of this solution, but it is one that come to mind first. Adding this next to `HKD` definition:
```haskell
class HKDFunctor f where
  hkdMap :: Proxy f -> (a -> b) -> HKD f a -> HKD f b

instance HKDFunctor Identity where
  hkdMap _ f a = f a

instance HKDFunctor Maybe where
  hkdMap _ f = fmap f

instance HKDFunctor StrictMaybe where
  hkdMap _ f = fmap f
```

Then I think implementation can be:
```haskell
translatePParams ::
  forall f c. HKDFunctor f => Alonzo.PParams f (AlonzoEra c) -> PParams' f (BabbageEra c)
translatePParams (Alonzo.PParams {_coinsPerUTxOWord = cpuw, ..}) = PParams {_coinsPerUTxOByte = cpub, ..}
  where
    cpub = hkMap (Proxy :: Proxy f) coinsPerUTxOWordToCoinsPerUTxOByte  cpuw
```

Reply:
ah cool! thinks it's worth it? :thinking:

Reply:
It might be worth it. It's up to you. It could prove handy again in the future eras.

Reply:
it worked great, I'm happy with it, thanks!